model/Image.php

```diff
@@ -1037,8 +1033,9 @@ class Image_Cached extends Image {
 	 * @param boolean $isSingleton This this to true if this is a singleton() object, a stub for calling methods.
 	 *                             Singletons don't have their defaults set.
 	 */
-	public function __construct($filename = null, $isSingleton = false) {
+	public function __construct($filename = null, $isSingleton = false, Image $sourceImage = null) {
 		parent::__construct(array(), $isSingleton);
+		if ($sourceImage) $this->update($sourceImage->toMap());
```

Comment:
It seems to cause a problem if you swap out the Image class using the injector. Couldn't say why exactly - but code above fixes the problem. Seems like a safe change so will open a PR on 3 if you agree?

The whole Image_Cached model is a bit awkward right now but think you've completely overhauled that for 4.0? :D